ambry-server/src/main/java/com/github/ambry/server/mysql/AccountStatsMySqlStore.java

```diff
@@ -0,0 +1,223 @@
+   * @param registry The {@link MetricRegistry}.
+   * @throws SQLException
+   */
+  public AccountStatsMySqlStore(List<MySqlUtils.DbEndpoint> dbEndpoints, String localDatacenter, String clustername,
+      String hostname, String localBackupFilePath, HostnameHelper hostnameHelper, MetricRegistry registry)
+      throws SQLException {
+    this(new MySqlDataAccessor(dbEndpoints, localDatacenter, new MySqlMetrics(AccountStatsMySqlStore.class, registry)),
+        clustername, hostname, localBackupFilePath, hostnameHelper, registry);
+  }
+
+  /**
+   * Constructor to create link {@link AccountStatsMySqlStore}. It's only used in tests.
+   * @param dataAccessor The {@link MySqlDataAccessor}.
+   * @param clustername  The name of the cluster, like Ambry-prod.
+   * @param hostname The name of the host.
+   * @param localBackupFilePath The filepath to local backup file.
+   * @param hostnameHelper The {@link HostnameHelper} to simplify the hostname.
+   * @param registry The {@link MetricRegistry}.
+   */
+  AccountStatsMySqlStore(MySqlDataAccessor dataAccessor, String clustername, String hostname,
+      String localBackupFilePath, HostnameHelper hostnameHelper, MetricRegistry registry) {
+    mySqlDataAccessor = dataAccessor;
+    accountReportsDao = new AccountReportsDao(dataAccessor, clustername, hostname);
+    this.hostnameHelper = hostnameHelper;
+    storeMetrics = new AccountStatsMySqlStore.Metrics(registry);
+    if (!Strings.isNullOrEmpty(localBackupFilePath)) {
+      // load backup file and this backup is the previous stats
+      ObjectMapper objectMapper = new ObjectMapper();
+      try {
+        this.previousStats = objectMapper.readValue(new File(localBackupFilePath), StatsWrapper.class);
+      } catch (Exception e) {
+        this.previousStats = null;
+      }
+    }
+  }
+
+  /**
+   * Publish the {@link StatsWrapper} to mysql database. This method ignores the error information from {@link StatsWrapper}
+   * and only publish the container storage usages that are different from the previous one.
+   * @param statsWrapper The {@link StatsWrapper} to publish.
+   */
+  public void publish(StatsWrapper statsWrapper) {
+    StatsSnapshot prevSnapshot =
+        previousStats == null ? new StatsSnapshot((long) -1, new HashMap<>()) : previousStats.getSnapshot();
+    applyFunctionToContainerUsageInDifferentStatsSnapshots(statsWrapper.getSnapshot(), prevSnapshot,
+        this::tryUpdateStorageUsage);
+    previousStats = statsWrapper;
+  }
+
+  /**
+   * Query mysql database to get all the container storage usage for given {@code clustername} and {@code hostname} and
+   * construct a {@link StatsSnapshot} from them.
+   * @param clustername the clustername.
+   * @param hostname the hostname
+   * @return {@link StatsSnapshot} published by the given host.
+   * @throws SQLException
+   */
+  public StatsSnapshot queryStatsSnapshotOf(String clustername, String hostname) throws SQLException {
```

Comment:
Looks like this method is called in tests only?

Reply:
it's now, but in my next PR, to implement mysql database aggregation task, it will be used to query the stats from a given host.